### PR TITLE
Improve parsing of PyTorch functional layers arguments

### PR DESCRIPTION
**Pull request by Idan-BenAmi**
## Pull Request Description:
Support PyTorch nn.functional.layer_norm and add a relevant test.

## Checklist before requesting a review:
- [x] I set the appropriate labels on the pull request.
- [x] I have added/updated the release note draft (if necessary).
- [x] I have updated the documentation to reflect my changes (if necessary).
- [x] All function and files are well documented. 
- [x] All function and classes have type hints.
- [x] There is a licenses in all file.
- [x] The function and variable names are informative. 
- [x] I have checked for code duplications.
- [x] I have added new unittest (if necessary).

### CODE REVIEW

File: tests/pytorch_tests/model_tests/feature_models/layer_norm_net_test.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
+# Copyright 2022 Sony Semiconductor Israel, Inc. All rights reserved.
```
Comment (haihabi) [inline tests/pytorch_tests/model_tests/feature_models/layer_norm_net_test.py:1]: The year is 2024

File: model_compression_toolkit/core/pytorch/graph_substitutions/substitutions/functional_layer_norm.py [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+            Graph after applying the substitution.
+        """
+        normalized_shape = node.input_shape[0][-1]
+
+        ln_node_weights = self.get_attributes_from_inputs(graph, node, normalized_shape)
+        if not ln_node_weights:
+            return graph
+        new_layernorm = BaseNode(name=node.name + '_into_LayerNorm',
+                                   framework_attr={NORMALIZED_SHAPE: normalized_shape,
+                                                   EPSILON: EPSILON_VAL,
```
Comment (lior-dikstein) [inline model_compression_toolkit/core/pytorch/graph_substitutions/substitutions/functional_layer_norm.py:90]: epsilon_val: why not get this from the node attribute?

File: model_compression_toolkit/core/pytorch/graph_substitutions/substitutions/functional_layer_norm.py [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+            node: node that match the pattern in the substitution init.
+
+        Returns:
+            Graph after applying the substitution.
+        """
+        normalized_shape = node.input_shape[0][-1]
+
+        ln_node_weights = self.get_attributes_from_inputs(graph, node, normalized_shape)
+        if not ln_node_weights:
+            return graph
```
Comment (lior-dikstein) [inline model_compression_toolkit/core/pytorch/graph_substitutions/substitutions/functional_layer_norm.py:87]: why do you need this? ln_node_weights is always a dict

**Review by lior-dikstein — APPROVED** [review]
